File: CSCI_313-323_Data_Structures_And_Algorithms/CSCI_313_Data_Structures/ExCode/02-IterativeAlgorithms/LeetCode-0771-1.cpp
class Solution {
public:
    ///------------------------------------------------------------------
    /// This is O(n*m) algorithm: For each char in S, search it in J.
    /// If it exists in J, then increment the count
    ///
    int numJewelsInStones(string jewels, string stones) {
        int count = 0;
        for (auto ch1: stones){
            for (auto ch2: jewels){
                if (ch1 == ch2){
                    count++;
                    break;
                } // end-if
            } //end-for
        } // end-for

        return count;        
    } // end-numJewelsInStones
};

File: CSCI_313-323_Data_Structures_And_Algorithms/CSCI_313_Data_Structures/ExCode/02-IterativeAlgorithms/LeetCode-0771-1.py
class Solution:
    # This is O(n*m) algorithm: For each char in S, search it in J.
    # If it exists in J, then increment the count
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        count = 0
        for ch1 in stones:
            if ch1 in jewels:
                    count += 1
        return count

